packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNodeUtils.ts

```diff
@@ -64,104 +64,46 @@ export interface ISummarizerNodeRootContract {
-			referenceSequenceNumber,
-			basePath: undefined,
-			localPath: EscapedPath.create(""), // root hard-coded to ""
-		});
-	}
-
-	/** Summary reference sequence number, i.e. last sequence number seen when it was created */
-	public get referenceSequenceNumber(): number {
-		return this.summary.referenceSequenceNumber;
-	}
-	/** Full path to parent node, or undefined if this is the root */
-	public get basePath(): EscapedPath | undefined {
-		return this.summary.basePath;
-	}
-	/** Relative path to this node from its parent node */
-	public get localPath(): EscapedPath {
-		return this.summary.localPath;
-	}
-	/** Relative path from this node to its node innermost base summary */
-	public get additionalPath(): EscapedPath | undefined {
-		return this.summary.additionalPath;
-	}
-	public set additionalPath(additionalPath: EscapedPath | undefined) {
-		this.summary.additionalPath = additionalPath;
-	}
-	constructor(
-		private readonly summary: {
-			readonly referenceSequenceNumber: number;
-			readonly basePath: EscapedPath | undefined;
-			readonly localPath: EscapedPath;
-			additionalPath?: EscapedPath;
-		},
-	) {}
-
-	/** Gets the full path to this node, to be used when sending a handle */
-	public get fullPath(): EscapedPath {
-		return this.basePath?.concat(this.localPath) ?? this.localPath;
-	}
-
-	/**
-	 * Gets the full path to this node's innermost base summary.
-	 * The children nodes can use this as their basePath to determine their path.
-	 */
-	public get fullPathForChildren(): EscapedPath {
-		return this.additionalPath !== undefined
-			? this.fullPath.concat(this.additionalPath)
-			: this.fullPath;
-	}
-
 	/**
-	 * Creates a new node within the same summary for a child of this node.
-	 * @param id - id of the child node
+	 * Creates and returns a new instance of this class for child of the current node.
 	 */
-	public createForChild(id: string): SummaryNode {
-		return new SummaryNode({
-			referenceSequenceNumber: this.referenceSequenceNumber,
-			basePath: this.fullPathForChildren,
-			localPath: EscapedPath.create(id),
-		});
+	public createChildPath(childNodePath: EscapedPath): EscapedPath {
+		return new EscapedPath(
+			`${this.path}/${encodeURIComponent(channelsTreeName)}/${childNodePath.path}`,
+		);
 	}
 }
+export interface PendingSummaryInfo {
+	/** The sequence number up until which the summary was created. */
+	referenceSequenceNumber: number;
+}
 
 /**
  * Represents the details needed to create a child summarizer node.
  */
 export interface ICreateChildDetails {
-	/** Latest summary from server node data */
-	latestSummary: SummaryNode | undefined;
 	/** Sequence number of latest known change to the node */
 	changeSequenceNumber: number;
 	/** A unique id of this child to be logged when sending telemetry. */
 	telemetryNodeId: string;
+	/** Summary handle for child node */
+	summaryHandleId: EscapedPath;
+	/** last reference sequence number seen when last successful summary was created */
```

Comment:
This doesn't seem quite right. It is more like the the reference sequence number of the last successful summary.